### ISSUE TITLE
Integration with the /api/collections endpoint? 

### ISSUE THREAD

**Issue by unaidedelf8777** [huggingface/huggingface.js#271]
Hi,

I am currently working on a website for my org, and I was really hoping to be able to make a collection of our best models and datasets, so that they can be displayed on our landing page. However, Right now it seems that I will have to fetch the data from the collections API from a local site file. this also means that it will not be realtime, and I will have to manually update the websites files each time I want to update the site, which isn't ideal. It would be awesome if you guys could integrate the collections into the huggingface/hub.js library. 

regards.

**Comment by julien-c** [comment]
Cool idea, would be great to have support for this in the `@huggingface/hub` library

I'd adding a https://github.com/huggingface/huggingface.js/labels/good%20first%20issue label, anyone from the community wants to give it a try to implement it here?

Collection API is documented here BTW: https://huggingface.co/docs/hub/api#collection-api

<img width="1074" alt="image" src="https://github.com/huggingface/huggingface.js/assets/326577/dc57063f-0e18-4e07-92da-7e59e5f504b0">

(cc @xenova for visibility too)

**Comment by Wauplin** [comment]
As additional resource, here is how it is implemented in the Python client: https://github.com/huggingface/huggingface_hub/blob/b3e06ae2f01b30b9ff9fe391245d56b0336b048c/src/huggingface_hub/hf_api.py#L5900. Might be worth taking a look while implementing the JS counterpart :wink:

**Comment by hackpk** [comment]
@julien-c and @Wauplin Can I work on this??

**Comment by Wauplin** [comment]
@hackpk Yes, please go ahead! Thanks for proposing to contribute on this :hugs:
If you have any question about the API, please let us know. Once you have a PR to review, please ping @coyotte508 for review. Looking forward to it!

**Comment by hackpk** [comment]
hi, @Wauplin and @coyotte508 I have worked on Sap Cds projects and it would be helpful if you could tell me how can I run and debug the code I have written in hub/src/lib to get the collections.

**Comment by hackpk** [comment]
Hi @coyotte508 is there any solution for this error
![image](https://github.com/huggingface/huggingface.js/assets/22720175/49a24082-1d81-4638-bb05-c6279199e12b)

Tried login with huggingface_hub login token but nothing.

**Comment by robert-j-webb** [comment]
Hey any update on this?

**Comment by julien-c** [comment]
i think this has been implemented by #1568! Yay!
